packages/react/src/ButtonGroup/ButtonGroup.module.css

```diff
@@ -0,0 +1,8 @@
+.ButtonGroup {
+  display: inline-grid;
+  gap: 8px;
+}
+
+.ButtonGroup--horizontal {
```

Comment:
Could direction be controlled responsively instead of explicitly through props?

I would have assumed that buttons should always stack vertically in narrow viewports and span horizontally on larger ones. 

Right now, this happens on the narrow viewports 👇 
![two super squished buttons in a narrow viewport](https://user-images.githubusercontent.com/13340707/199045563-b24f6714-1fd2-4689-ac06-1569a5eed9c1.png)

If we keep this approach of direction via a prop, but still wanted the functionality I mentioned it would force app-side devs to have a matchMedia listener, which probably isn't ideal. 

cc. @ashygee also for any design considerations.